solutions/CF Mixing Water.mdx

```diff
@@ -0,0 +1,118 @@
+---
+id: cf-1359C
+source: CF
+title: Mixing Water
+author: Jesse Choe
+---
+
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- If the number of cups of hot water is equal to the number of cups of cold water in the barrel, then the minimum number of cups required to pour into the barrel is exactly $2$.
+- The number of cups of hot water must be greater than the number of cups of cold water if the previous condition isn't satisfied. Thus, there will be $c + 1$ cups of hot water and $c$ cups of cold water.
+- The two barrels with the closest temperature to $t$ will be the barrels with temperatures slightly more and slightly less than $t$. The two barrels will be $a$ and $b$, respectively.
+
+Binary search on the maximum number of cups $x$ with a temperature of at least $t$.
+
+$a$ will consist of $x$ cups poured, and $b$ will consist of $x + 2$ cups poured (from the second observation listed above, $x$ is guaranteed to be odd).
+
+The answer will be the barrel with a minimum $|t_b - t|$, which can either be the barrel with $2$, $x$, or $x + 2$ cups.
+
+**Time Complexity:** $\mathcal{O}(\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+double hot_water, cold_water, desired_temp;
+
+/*
+	* This function checks whether pouring cupsPoured amount of cups achieves an average
+	* barrel temperature greater than or equal to desired_temp. Technically, this represents
+	* 2 * cupsPoured + 1 cups
+*/
+
+bool works(int cupsPoured) {
+	double totalBarrelTemperature = (hot_water * (cupsPoured + 1) + cold_water * (cupsPoured));
+	double totalCupsPoured = (double) (2 * cupsPoured + 1.0);
+	double avgBarrelTemperature = totalBarrelTemperature / totalCupsPoured;
+	return avgBarrelTemperature >= desired_temp;
+}
+
+void solve() {
+	/* 
+	   * maximumCups represents the the maximum number of cups such
+	   * that the barrel temperature is greater than desired_temp.
+	*/
+	cin >> hot_water >> cold_water >> desired_temp;
+	// Binary search on the answer for maximumCups.
+	int left = 0, right = 1e6 + 1, maximumCups = 0;
+
+	while(left <= right) {
```

Comment:
maybe also spaces between the control statements and the parens? (`while (left <= right)`)?

Reply:
as long as it's consistent it's fine.